File: src/main/java/net/greenjab/fixedminecraft/registry/item/GlisteringMelonSliceItem.java
package net.greenjab.fixedminecraft.registry.item;

import net.minecraft.entity.LivingEntity;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.server.world.ServerWorld;
import net.minecraft.sound.SoundCategory;
import net.minecraft.sound.SoundEvent;
import net.minecraft.sound.SoundEvents;
import net.minecraft.util.math.GlobalPos;
import net.minecraft.util.math.Vec3d;
import net.minecraft.world.World;
import net.minecraft.world.event.GameEvent;

import java.util.Optional;
import java.util.Set;

public class GlisteringMelonSliceItem extends Item {
    public GlisteringMelonSliceItem(Settings settings) {
        super(settings);
    }

    @Override
    public ItemStack finishUsing(ItemStack stack, World world, LivingEntity user) {
        super.finishUsing(stack, world, user);
        if (user instanceof ServerPlayerEntity serverPlayerEntity) {
            serverPlayerEntity.setHealth(user.getHealth()+2);
        }
        return stack;
    }
}

File: src/main/kotlin/net/greenjab/fixedminecraft/registry/item/GlisteringMelonSliceItem.kt
package net.greenjab.fixedminecraft.registry.item

import net.minecraft.entity.LivingEntity
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class GlisteringMelonSliceItem(settings: Settings?) : Item(settings) {
    override fun finishUsing(stack: ItemStack, world: World, user: LivingEntity): ItemStack {
        super.finishUsing(stack, world, user)
        if (!world.isClient) {
            user.health+=2
        }
        return stack
    }
}
